File: back/src/main/java/com/example/back/dto/SavedRouteResponseDTO.java
package com.example.back.dto;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SavedRouteResponseDTO {
    private boolean success;
    private String message;
    private String savedRouteName; // 이미 존재하는 경우 해당 이름 반환
    private boolean alreadyExists; // Flag to indicate if the route already existed

}

File: front/app/src/main/java/com/example/front/transportation/data/DB/SavedRouteResponseDTO.kt
package com.example.front.transportation.data.DB

import com.google.gson.annotations.SerializedName

data class SavedRouteResponseDTO(
    @SerializedName("success") val success: Boolean,
    @SerializedName("message") val message: String,
    @SerializedName("savedRouteName") val savedRouteName: String?, // 서버 응답에 'savedRouteId'가 없음 (선택적)
    @SerializedName("savedRouteId") val savedRouteId: Int?,       // 서버 응답에 'savedRouteId'가 포함될 수 있음 (선택적)
    @SerializedName("alreadyExists") val alreadyExists: Boolean
)
